### ISSUE TITLE
Error reading IAM Role in terraform destroy

### ISSUE THREAD

**Issue by amandasoaares** [localstack/localstack#2286]
Hi guys, how you are doing?

I working in some tests to terraform with terratest and localstack but I get some problems with to destroy resources IAM Role and IAM Policy

```
provider "aws" {
  access_key                  = "mock_access_key"
  region                      = local.origin_region
  s3_force_path_style         = true
  secret_key                  = "mock_secret_key"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true


  endpoints {
    apigateway     = "http://localstack-s3:4567"
    cloudformation = "http://localstack-s3:4581"
    cloudwatch     = "http://localstack-s3:4582"
    dynamodb       = "http://localstack-s3:4569"
    es             = "http://localstack-s3:4578"
    firehose       = "http://localstack-s3:4573"
    iam            = "http://localstack-s3:4593"
    kinesis        = "http://localstack-s3:4568"
    kms            = "http://localstack-s3:4599"
    lambda         = "http://localstack-s3:4574"
    route53        = "http://localstack-s3:4580"
    redshift       = "http://localstack-s3:4577"
    s3             = "http://localstack-s3:4572"
    secretsmanager = "http://localstack-s3:4584"
    ses            = "http://localstack-s3:4579"
    sns            = "http://localstack-s3:4575"
    sqs            = "http://localstack-s3:4576"
    ssm            = "http://localstack-s3:4583"
    stepfunctions  = "http://localstack-s3:4585"
    sts            = "http://localstack-s3:4592"
  }
}

provider "aws" {
  access_key                  = "mock_access_key"
  region                      = local.replica_region
  s3_force_path_style         = true
  secret_key                  = "mock_secret_key"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  alias                       = "replica"

  endpoints {
    apigateway     = "http://localstack-s3:4567"
    cloudformation = "http://localstack-s3:4581"
    cloudwatch     = "http://localstack-s3:4582"
    dynamodb       = "http://localstack-s3:4569"
    es             = "http://localstack-s3:4578"
    firehose       = "http://localstack-s3:4573"
    iam            = "http://localstack-s3:4593"
    kinesis        = "http://localstack-s3:4568"
    kms            = "http://localstack-s3:4599"
    lambda         = "http://localstack-s3:4574"
    route53        = "http://localstack-s3:4580"
    redshift       = "http://localstack-s3:4577"
    s3             = "http://localstack-s3:4572"
    secretsmanager = "http://localstack-s3:4584"
    ses            = "http://localstack-s3:4579"
    sns            = "http://localstack-s3:4575"
    sqs            = "http://localstack-s3:4576"
    ssm            = "http://localstack-s3:4583"
    stepfunctions  = "http://localstack-s3:4585"
    sts            = "http://localstack-s3:4592"
  }
}

data "aws_caller_identity" "current" {}

resource "random_pet" "this" {
  length = 2
}

resource "aws_kms_key" "replica" {
  provider = aws.replica

  description             = "S3 bucket replication KMS key"
  deletion_window_in_days = 7
}

module "replica_bucket" {
  source = "../../"

  providers = {
    aws = aws.replica
  }

  bucket = local.destination_bucket_name
  region = local.replica_region
  acl    = "private"

  versioning = {
    enabled = true
  }
}

module "s3_bucket" {
  source = "../../"

  bucket = local.bucket_name
  region = local.origin_region
  acl    = "private"

  versioning = {
    enabled = true
  }

  replication_configuration = {
    role = aws_iam_role.replication.arn

    rules = [
      {
        id       = "foo"
        status   = "Enabled"
        priority = 10

        source_selection_criteria = {
          sse_kms_encrypted_objects = {
            enabled = true
          }
        }

        filter = {
          prefix = "one"
          tags = {
            ReplicateMe = "Yes"
          }
        }

        destination = {
          bucket             = "arn:aws:s3:::${local.destination_bucket_name}"
          storage_class      = "STANDARD"
          replica_kms_key_id = aws_kms_key.replica.arn
          account_id         = data.aws_caller_identity.current.account_id
          access_control_translation = {
            owner = "Destination"
          }
        }
      },
      {
        id       = "bar"
        status   = "Enabled"
        priority = 20

        destination = {
          bucket        = "arn:aws:s3:::${local.destination_bucket_name}"
          storage_class = "STANDARD"
        }


        filter = {
          prefix = "two"
          tags = {
            ReplicateMe = "Yes"
          }
        }

      },

    ]
  }

}
```

```
resource "aws_iam_role" "replication" {
  name                  = "s3-bucket-replication-${random_pet.this.id}"
  force_detach_policies = true


  assume_role_policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "s3.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
POLICY
}

resource "aws_iam_policy" "replication" {
  name = "s3-bucket-replication-${random_pet.this.id}"

  policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:GetReplicationConfiguration",
        "s3:ListBucket"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::${local.bucket_name}"
      ]
    },
    {
      "Action": [
        "s3:GetObjectVersion",
        "s3:GetObjectVersionAcl"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::${local.bucket_name}/*"
      ]
    },
    {
      "Action": [
        "s3:ReplicateObject",
        "s3:ReplicateDelete"
      ],
      "Effect": "Allow",
      "Resource": "arn:aws:s3:::${local.destination_bucket_name}/*"
    }
  ]
}
POLICY
}

resource "aws_iam_policy_attachment" "replication" {
  name       = "s3-bucket-replication-${random_pet.this.id}"
  roles      = [aws_iam_role.replication.name]
  policy_arn = aws_iam_policy.replication.arn
}
```

When I run the follow commands in my account aws it's work perfectly

```
terraform init
terraform validate
terraform apply
terraform destroy
```

But when I run in localstack the terraform apply works fine but in terraform destroy I get this errors:

```
Error: Error reading IAM Role s3-bucket-replication-gentle-mosquito: : 
go_1          | TestS3Simple 2020-04-12T02:51:33Z command.go:158:       status code: 404, request id:
```

any help would be appreciated
Thanks

**Comment by whummer** [comment]
Thanks for reporting @amandasoaares . This should be fixed in #2289 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks!

**Comment by amandasoaares** [comment]
> Thanks for reporting @amandasoaares . This should be fixed in #2289 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks!

It work @whummer  and @duongpv7 , Thank so much.
